proposals/XXXX-associated-types-constraints.md

```diff
@@ -0,0 +1,100 @@
+# Permit where clauses to constrain associated types
+
+* Proposal: [SE-XXXX](https://github.com/apple/swift-evolution/blob/master/proposals/XXXX-associated-types-constraints.md)
+* Author(s): [David Hart](http://github.com/hartbit), [Jacob Bandes-Storch](jtbandes@gmail.com), [Douglas Gregor](dgregor@apple.com)
+* Status: TBD
+* Review manager: TBD
+
+## Introduction
+
+This proposal seeks to introduce a `where` clause to associated types
+declarations and improvements to protocol constraints to bring associated types
+the same expressive power as generic types.
+
+This proposal was discussed twice on the Swift Evolution list in the following
+threads:
+
+- [[Completing Generics] Arbitrary requirements in protocols](http://thread.gmane.org/gmane.comp.lang.swift.evolution/14243)
+- [More Powerful Constraints for Associated Types](http://thread.gmane.org/gmane.comp.lang.swift.evolution/15201)
```

Comment:
Although the [Gmane web interface is back](https://lars.ingebrigtsen.no/2016/09/06/gmane-alive/) those links should be updated.

```markdown
* [\[Completing Generics\] Arbitrary requirements in protocols](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20160411/014667.html)
* [Proposal] More Powerful Constraints for Associated Types
  - [Week #1](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20160418/015625.html)
  - [Week #2](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20160425/015753.html)
  - [Week #3](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20160502/016354.html)
```